when build script fails, cargo gives no indication of which package failed

Example:

```
[jack@ocate cef]$ cargo build
    Updating git repository `https://github.com/servo/glfw-rs`
    Updating git repository `https://github.com/servo/libfontconfig`
    Updating git repository `https://github.com/servo/sharegl`
   Compiling mozjs-sys v0.0.0 (https://github.com/servo/mozjs#b6abbc29)
   Compiling harfbuzz v0.1.0 (https://github.com/servo/rust-harfbuzz#f07c6ae3)
   Compiling parserutils-sys v0.1.1 (http://github.com/servo/libparserutils#4148428a)
   Compiling macros v0.0.1 (file:///home/jack/src/servo/ports/cef)
   Compiling string_cache_macros v0.0.0 (https://github.com/servo/string-cache#aa4365ac)
   Compiling expat-sys v2.1.0 (https://github.com/servo/libexpat#da2ddaf7)
   Compiling freetype-sys v2.4.11 (https://github.com/servo/libfreetype2#5b649916)
   Compiling link_glfw v0.1.0 (file:///home/jack/src/servo/support/glfw-rs)
   Compiling geom v0.1.0 (https://github.com/servo/rust-geom#2982b770)
   Compiling stb_image v0.1.0 (https://github.com/servo/rust-stb-image#99679102)
   Compiling egl v0.1.0 (https://github.com/servo/rust-egl#48b85e30)
   Compiling xlib v0.1.0 (https://github.com/servo/rust-xlib#79904fb4)
   Compiling task_info v0.0.1 (file:///home/jack/src/servo/ports/cef)
   Compiling openssl v0.0.0 (https://github.com/sfackler/rust-openssl.git#cd40d25c)
   Compiling png-sys v1.6.3 (https://github.com/servo/libpng?ref=servo#7bc6e343)
   Compiling freetype v0.1.0 (https://github.com/servo/rust-freetype#fb15dd30)
   Compiling core_foundation v0.1.0 (https://github.com/servo/rust-core-foundation#9aeaee0b)
   Compiling semver v0.0.1 (https://github.com/rust-lang/semver#df163f7b)
   Compiling phf v0.0.0 (https://github.com/sfackler/rust-phf#fa5d8034)
   Compiling encoding v0.1.0 (https://github.com/lifthrasiir/rust-encoding#12b6610a)
   Compiling opengles v0.1.0 (https://github.com/servo/rust-opengles#13cccda1)
   Compiling cocoa v0.1.0 (https://github.com/servo/rust-cocoa#f5da9ccf)
   Compiling phf_mac v0.0.0 (https://github.com/sfackler/rust-phf#fa5d8034)
   Compiling skia-sys v0.0.20130412 (https://github.com/servo/skia#6d696712)
   Compiling fontconfig-sys v2.11.1 (https://github.com/servo/libfontconfig#1fc43065)
   Compiling hubbub-sys v0.1.2 (https://github.com/servo/libhubbub#b1dceaca)
   Compiling core_graphics v0.1.0 (https://github.com/servo/rust-core-graphics#04bd18a4)
   Compiling glfw v0.0.1 (file:///home/jack/src/servo/support/glfw-rs)
   Compiling alert v0.1.0 (https://github.com/servo/rust-alert#fdc24f13)
   Compiling io_surface v0.1.0 (https://github.com/servo/rust-io-surface#05f10227)
   Compiling glut v0.0.1 (https://github.com/servo/rust-glut#d351eed4)
Build failed, waiting for other jobs to finish...
Process didn't exit successfully: `make -f makefile.cargo` (status=2)
--- stderr
make: makefile.cargo: No such file or directory
make: *** No rule to make target `makefile.cargo'.  Stop.
```

--- a/src/cargo/ops/cargo_rustc/mod.rs
+++ b/src/cargo/ops/cargo_rustc/mod.rs
@@ -182,6 +182,7 @@ fn compile_custom(pkg: &Package, cmd: &str,
         p = p.env(format!("DEP_{}_OUT_DIR", name).as_slice(),
                   Some(&layout.native(pkg)));
     }
+    let pkg = pkg.to_string();
 
     Ok(proc() {
         if first {
@@ -193,7 +194,11 @@ fn compile_custom(pkg: &Package, cmd: &str,
                 internal("failed to create output directory for build command")
             }));
         }
-        try!(p.exec_with_output().map(|_| ()).map_err(|e| e.mark_human()));
+        try!(p.exec_with_output().map(|_| ()).map_err(|mut e| {
+            e.msg = format!("Failed to run custom build command for `{}`\n{}",
+                            pkg, e.msg);
+            e.mark_human()
+        }));
         Ok(())
     })
 }

--- a/tests/test_cargo_compile.rs
+++ b/tests/test_cargo_compile.rs
@@ -691,11 +691,13 @@ test!(custom_build_failure {
         "#);
     assert_that(p.cargo_process("build"),
                 execs().with_status(101).with_stderr(format!("\
+Failed to run custom build command for `foo v0.5.0 ({dir})
 Process didn't exit successfully: `{}` (status=101)\n\
 --- stderr\n\
 task '<main>' failed at 'nope', {filename}:2\n\
 \n\
-", build.bin("foo").display(), filename = format!("src{}foo.rs", path::SEP))));
+", build.bin("foo").display(), filename = format!("src{}foo.rs", path::SEP),
+   dir = p.url())));
 })
 
 test!(custom_second_build_failure {
@@ -752,11 +754,13 @@ test!(custom_second_build_failure {
         "#);
     assert_that(p.cargo_process("build"),
                 execs().with_status(101).with_stderr(format!("\
+Failed to run custom build command for `foo v0.5.0 ({dir})
 Process didn't exit successfully: `{}` (status=101)\n\
 --- stderr\n\
 task '<main>' failed at 'nope', {filename}:2\n\
 \n\
-", build2.bin("bar").display(), filename = format!("src{}bar.rs", path::SEP))));
+", build2.bin("bar").display(), filename = format!("src{}bar.rs", path::SEP),
+   dir = p.url())));
 })
 
 test!(custom_build_env_vars {

